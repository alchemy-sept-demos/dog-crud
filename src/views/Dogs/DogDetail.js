import React, { useState, useEffect } from 'react';
// import { useParams } from 'react-router-dom';
import Dog from '../../components/dogs/Dog';
import Loader from '../../components/Loader/Loader';
import { fetchDogById } from '../../services/dogs';

export default function DogDetail({
  match: {
    params: { id },
  },
}) {
  // const params = useParams();
  const [dog, setDog] = useState({});
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    const fetchData = async () => {
      // const resp = await fetchDogById(params.id);
      const resp = await fetchDogById(id);
      setDog(resp[0]);
      setLoading(false);
    };
    fetchData();
  }, [id]);
  return (
    <div>
      {loading && <Loader />}
      {!loading && <Dog {...dog} showDetail={true} />}
    </div>
  );
}
