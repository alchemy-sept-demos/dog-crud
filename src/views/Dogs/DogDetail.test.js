import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import DogDetail from './DogDetail';

it('should render an individual dog', async () => {
  // render the thing
  // OPTION 1 -- when using react router hooks (useParams)
  // const { container } = render(
  //   // MemoryRouter is needed because <Teams> uses the <Link> component
  //   <MemoryRouter initialEntries={['/dogs/4']}>
  //     <Route path="/dogs/:id" component={DogDetail} />
  //   </MemoryRouter>
  // );

  // OPTION 2 -- when using props.match.params.id
  const { container } = render(<DogDetail match={{ params: { id: 4 } }} />);
  // wait for stuff to load
  await screen.findByText('Meet Franny');

  // take a snapshot
  expect(container).toMatchSnapshot();
});
