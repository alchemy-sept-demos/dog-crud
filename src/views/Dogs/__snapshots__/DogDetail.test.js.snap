// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an individual dog 1`] = `
<div>
  <div>
    <div
      class="dog-detail"
    >
      <h3>
        Meet 
        Franny
      </h3>
      <img
        src="https://placedog.net/500?id=51"
      />
      <p>
        Franny
         is a(n) 
        4 year old
         
        French Bulldog
      </p>
      <div
        class="details"
      >
        <p>
          Sweet Frenchie with lots of love to give.
        </p>
      </div>
    </div>
  </div>
</div>
`;
